src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs

```diff
@@ -258,12 +258,12 @@ public PackageSpec Clone()
             else
             {
                 targetFrameworks = new List<TargetFrameworkInformation>(TargetFrameworks.Count);
-                targetFrameworks.AddRange(TargetFrameworks.Select(item => item.Clone()));
+                targetFrameworks.AddRange(TargetFrameworks);
             }
 
             return new PackageSpec(
                 targetFrameworks,
-                Dependencies?.Select(item => item.Clone()).ToList(),
+                Dependencies?.ToList(),
                 RuntimeGraph?.Clone(),
```

Comment:
>Clone [](http://example.com/codeflow?start=30&length=5)

A whole bunch of other cloning going on here, but those allocate far less, and this is already borderline too much for me to change.

Reply:
A lot of the other types are almost always empty, and a bunch of them are already Obsolete in the PackageSpec.